dask_expr/reductions.py

```diff
@@ -9,72 +11,141 @@
 )
 from dask.utils import M, apply
 
-from dask_expr.expr import Elemwise, Expr, Projection
+from dask_expr.expr import Blockwise, Elemwise, Expr, Projection
+
+##
+## Base MapReduce classes
+##
 
 
-class ApplyConcatApply(Expr):
-    """Perform reduction-like operation on dataframes
+class MapReduce(Expr):
+    """Perform a map-reduction operation on a dataframe
```

Comment:
Sure - I also don't consider the `MapReduce` name to be the right fit. However, the current `ApplyConcatApply` name is also wrong, since we aren't actually concatenating unless that logic is specified in `combine`/`aggregate`.